lib/ansible/modules/lineinfile.py

```diff
@@ -44,6 +44,18 @@
     type: str
     aliases: [ regex ]
     version_added: '1.7'
+  literal:
```

Comment:
This should probably be called `literal_search` to better communicate it is used for searching.
```suggestion
  literal_search:
```

Reply:
@samdoran this change affects to all the documentation and `argument_spec`. I think is better to keep the `literal` name and document properly. What do you think?

Reply:
I don't think `literal` is self-explanatory enough. It could also be interpreted as a boolean value.

Regular expressions are almost always used for searching, so `regexp_search` would be  redundant.

Maybe `search_string` is a better name? Naming things is hard. 😕

Reply:
Let's go for `search_string`, 😆 . I will change all the references and retest. Thank you!